anaconda_anon_usage/utils.py

```diff
@@ -2,18 +2,50 @@
 import os
 import sys
 from os.path import dirname, exists
+from threading import RLock
 
-DEBUG = bool(os.environ.get("ANACONDA_ANON_USAGE_DEBUG"))
+DPREFIX = os.environ.get("ANACONDA_ANON_USAGE_DEBUG_PREFIX") or ""
+DEBUG = bool(os.environ.get("ANACONDA_ANON_USAGE_DEBUG")) or DPREFIX
+
+
+# While lru_cache is thread safe, it does not prevent two threads
+# from beginning the same computation. This simple cache mechanism
+# uses a lock to ensure that only one thread even attempts.
+CACHE = {}
```

Comment:
I'm wondering now, do we expect long-running software that includes this package, which could run out of memory when caching the various tokens? E.g. is Navigator making requests via a sub process or in-process?

Reply:
This is only going to hold the tokens for a single transaction. Navigator is running a subprocess